src/wp-includes/theme.php

```diff
@@ -1540,6 +1548,8 @@ function get_uploaded_header_images() {
 		}
 	}
 
+	set_transient( $transient_key, $header_images, HOUR_IN_SECONDS );
```

Comment:
I think this is something that would be reasonable to cache for longer than 1 hour, e.g. 1 day at least.

If the theme is switched, this will automatically be invalidated anyway, and the other crucial piece (see my overarching review feedback) is that we'll need to invalidate when such an image is newly uploaded.

Reply:
set top a day